core/src/main/scala/org/ensime/core/Analyzer.scala

```diff
@@ -51,17 +52,22 @@ class Analyzer(
     broadcaster: ActorRef,
     indexer: ActorRef,
     search: SearchService,
+    scoped: List[EnsimeProjectId],
```

Comment:
Yes, that'll be safer. But `config` should also have `projects` on which the `scoped` projects depend.